### PR TITLE
Vendor 0.40

### PR DESCRIPTION
**Pull request by jeremydvoss**

### CODE REVIEW

File: azure-monitor-opentelemetry/azure/monitor/opentelemetry/_vendor/v0_40b0/opentelemetry/instrumentation/asgi/__init__.py [hunk 1]
```diff
@@ -242,18 +415,23 @@ def set_status_code(span, status_code):
 
 
 def get_default_span_details(scope: dict) -> Tuple[str, dict]:
-    """Default implementation for get_default_span_details
+    """
+    Default span name is the HTTP method and URL path, or just the method.
+    https://github.com/open-telemetry/opentelemetry-specification/pull/3165
+    https://opentelemetry.io/docs/reference/specification/trace/semantic_conventions/http/#name
+
```
Comment (jeremydvoss) [inline azure-monitor-opentelemetry/azure/monitor/opentelemetry/_vendor/v0_40b0/opentelemetry/instrumentation/asgi/__init__.py:422]: Check this change

File: azure-monitor-opentelemetry/azure/monitor/opentelemetry/_vendor/v0_40b0/opentelemetry/instrumentation/dbapi/__init__.py [hunk 2]
```diff
@@ -12,6 +12,30 @@
...
+
+
+    # Ex: mysql.connector
+    trace_integration(mysql.connector, "connect", "mysql")
+    # Ex: pyodbc
+    trace_integration(pyodbc, "Connection", "odbc")
+
+API
+---
+"""
```
Comment (jeremydvoss) [inline azure-monitor-opentelemetry/azure/monitor/opentelemetry/_vendor/v0_40b0/opentelemetry/instrumentation/dbapi/__init__.py:38]: remove docstring

File: azure-monitor-opentelemetry/azure/monitor/opentelemetry/_vendor/v0_40b0/opentelemetry/instrumentation/asgi/__init__.py [hunk 3]
```diff
@@ -13,6 +13,181 @@
...
+    export OTEL_INSTRUMENTATION_HTTP_CAPTURE_HEADERS_SANITIZE_FIELDS=".*session.*,set-cookie"
+
+will replace the value of headers such as ``session-id`` and ``set-cookie`` with ``[REDACTED]`` in the span.
+
+Note:
+    The environment variable names used to capture HTTP headers are still experimental, and thus are subject to change.
+
+API
+---
+"""
```
Comment (jeremydvoss) [inline azure-monitor-opentelemetry/azure/monitor/opentelemetry/_vendor/v0_40b0/opentelemetry/instrumentation/asgi/__init__.py:190]: remove docstring

File: azure-monitor-opentelemetry/azure/monitor/opentelemetry/_vendor/v0_40b0/opentelemetry/instrumentation/django/__init__.py [hunk 4]
```diff
@@ -0,0 +1,380 @@
...
+::
+
+    export OTEL_INSTRUMENTATION_HTTP_CAPTURE_HEADERS_SANITIZE_FIELDS=".*session.*,set-cookie"
+
+will replace the value of headers such as ``session-id`` and ``set-cookie`` with ``[REDACTED]`` in the span.
+
+Note:
+    The environment variable names used to capture HTTP headers are still experimental, and thus are subject to change.
+
+API
```
Comment (jeremydvoss) [inline azure-monitor-opentelemetry/azure/monitor/opentelemetry/_vendor/v0_40b0/opentelemetry/instrumentation/django/__init__.py:233]: remove docstring

File: azure-monitor-opentelemetry/azure/monitor/opentelemetry/_vendor/v0_40b0/opentelemetry/instrumentation/fastapi/__init__.py [hunk 5]
```diff
@@ -0,0 +1,364 @@
...
+::
+
+    export OTEL_INSTRUMENTATION_HTTP_CAPTURE_HEADERS_SANITIZE_FIELDS=".*session.*,set-cookie"
+
+will replace the value of headers such as ``session-id`` and ``set-cookie`` with ``[REDACTED]`` in the span.
+
+Note:
+    The environment variable names used to capture HTTP headers are still experimental, and thus are subject to change.
+
+API
```
Comment (jeremydvoss) [inline azure-monitor-opentelemetry/azure/monitor/opentelemetry/_vendor/v0_40b0/opentelemetry/instrumentation/fastapi/__init__.py:171]: remove dosctring

File: azure-monitor-opentelemetry/azure/monitor/opentelemetry/_vendor/v0_40b0/opentelemetry/instrumentation/asgi/__init__.py (hunk 1 above)
Comment (jeremydvoss) [reply azure-monitor-opentelemetry/azure/monitor/opentelemetry/_vendor/v0_40b0/opentelemetry/instrumentation/asgi/__init__.py:247]: This seems like a potential breaking change:
https://github.com/open-telemetry/opentelemetry-python-contrib/pull/1759

Comment (jeremydvoss) [reply azure-monitor-opentelemetry/azure/monitor/opentelemetry/_vendor/v0_40b0/opentelemetry/instrumentation/asgi/__init__.py:247]: Checking results of requests and dependencies. All columns match. So, I think we can consider this change to be okay.

**Comment by lzchen** [comment]
Any reason we want to support 0.40b0 so soon? Are you planning on vendoring everytime there's a new release of the instrumentations?

**Comment by jeremydvoss** [comment]
> Any reason we want to support 0.40b0 so soon? Are you planning on vendoring everytime there's a new release of the instrumentations?

I guess being behind on vendoring doesn't block Cxs like being behind on the sdk/api. So, we can hold off.

**Comment by lzchen** [comment]
@jeremydvoss 

I think the key is to have an understanding between us and the customers. Should we only update it when there's a release, or should it be update only if there are big/breaking/feature changes? I think we can start with the latter and if there is sufficient demand for latest features we can change our process.

**Comment by jeremydvoss** [comment]
Azure SDL Version: https://github.com/Azure/azure-sdk-for-python/pull/31719

**Comment by jeremydvoss** [comment]
Closing in favor of unvendoring: https://github.com/Azure/azure-sdk-for-python/pull/31744

**Pull request closed without merging**